src/js/app.js

```diff
@@ -53,123 +54,164 @@ Promise.prototype.always = function(onAlways) {
-          self.drive.authorize();
-        }).fail(function() {
-          alert("Unable to sign in.");
-        });
-      }});
-
-      self.drive.onStateChange(function(state) {
-        if (state == App.Drive.State.UNKNOWN) {
-
-          self.logging.info("Google Drive state unknown.");
-
-        } else if (state == App.Drive.State.UNAUTHORIZED) {
-
-          self.logging.info("Google Drive state unauthorized.");
-          self.settingsButton.hide();
-          self.consoleButton.hide();
-          self.clear();
-          self.console.hide();
+      self.store = new App.Store("save-state", 50);
+      var initCallback = self.initCallback.bind(self);
+      self.store.open(initCallback);
+    },
 
-          if (window.navigator.onLine === true) {
-            self.drive.authURL().then(function(url) {
-              $("#google-drive-auth").attr("href", url);
-              $("#redeem-code").val('');
-              $("#screen-account").show();
-            }).fail(function() {
-              alert("Unable to generate Google authentication URL.");
-            });
+    initCallback: function(opened, error) {
+      var self = this;
+      if (opened) {
+        self.logging = new App.Logging(window.config.logging_level, "app");
+        self.logging.info("Version: " + window.config.version);
+        self.logging.info(
+          "Screen size: " + $(window).width() + " x " + $(window).height()
+        );
+        self.logging.info("User Agent: " + navigator.userAgent);
+
+        self.library = new App.Library(
+          self.store,
+          function(identifier) {
+            self.console.clear();
+            self.console.show();
+            setTimeout(function() {
+              self.load(identifier);
+            }, 400);
+          },
+          function() {
+            self.gameBoy = new App.GameBoy(self.store, self.library);
+            self.games = new App.Games(self.device, self.gameBoy, self.library);
+
+            self.console = new App.Console(
+              self.device,
+              self.gameBoy,
+              {
+                didHide: function() {
+                  self.games.update();
+                }
+              },
+              self.store
+            );
+            var callback = function() {
+              self.settings = new App.Settings(
+                self.drive,
+                self.store,
+                self.gameBoy,
+                self.console
+              );
+
+              self.settingsButton = new App.Controls.Button(
+                $("#button-account"),
+                {
+                  touchUpInside: function() {
+                    self.settings.show();
+                  }
+                }
+              );
+
+              self.consoleButton = new App.Controls.Button($("#button-done"), {
+                touchUp: function() {
+                  self.logging.info("Show console");
+                  self.console.show();
+                }
+              });
```

Comment:
You might be right, but I think this would be best covered in another issue/PR more focused on improving the codebase in general. I think the only modification I made was to include it in the callback function.